src/v/cloud_storage/topic_mount_handler.cc

```diff
@@ -0,0 +1,135 @@
+  const topic_mount_manifest& manifest,
+  const remote_path_provider& path_provider,
+  retry_chain_node& parent) {
+    const auto manifest_path = cloud_storage_clients::object_key{
+      manifest.get_manifest_path(path_provider)};
+
+    const auto exists_result = co_await _remote.object_exists(
+      _bucket, manifest_path, parent, existence_check_type::manifest);
+
+    // If the manifest exists, it is possible to mount the topic.
+    co_return (exists_result == download_result::success)
+      ? topic_mount_result::success
+      : topic_mount_result::mount_manifest_does_not_exist;
+}
+
+ss::future<topic_mount_result> topic_mount_handler::commit_mount(
+  const topic_mount_manifest& manifest,
+  const remote_path_provider& path_provider,
+  retry_chain_node& parent) {
+    const auto manifest_path = cloud_storage_clients::object_key(
+      manifest.get_manifest_path(path_provider));
+
+    const auto delete_result = co_await _remote.delete_object(
+      _bucket, manifest_path, parent);
+
+    co_return (delete_result == upload_result::success)
+      ? topic_mount_result::success
+      : topic_mount_result::mount_manifest_not_deleted;
+}
+
+ss::future<topic_mount_result> topic_mount_handler::mount_topic(
+  const remote_label& label,
+  const cluster::topic_configuration& topic_cfg,
+  retry_chain_node& parent) {
+    const auto remote_tp_ns = topic_cfg.remote_tp_ns();
+    const auto manifest = topic_mount_manifest(
+      label.cluster_uuid, remote_tp_ns);
+    const auto path_provider = remote_path_provider(label, remote_tp_ns);
+
+    const auto check_result = co_await check_mount(
+      manifest, path_provider, parent);
+    if (check_result != topic_mount_result::success) {
+        vlog(
+          cst_log.warn,
+          "Couldn't mount topic {}, check result was {}.",
+          topic_cfg.tp_ns,
+          check_result);
+        co_return check_result;
+    }
+
+    const auto commit_result = co_await commit_mount(
+      manifest, path_provider, parent);
+
+    if (commit_result != topic_mount_result::success) {
+        vlog(
+          cst_log.warn,
+          "Couldn't mount topic {}, commit result was {}.",
+          topic_cfg.tp_ns,
+          commit_result);
+    }
+
+    co_return commit_result;
+}
+
+ss::future<topic_unmount_result> topic_mount_handler::unmount_topic(
+  const remote_label& label,
+  const cluster::topic_configuration& topic_cfg,
+  retry_chain_node& parent) {
+    const auto remote_tp_ns = topic_cfg.remote_tp_ns();
+    const auto manifest = topic_mount_manifest(
+      label.cluster_uuid, remote_tp_ns);
+    const auto path_provider = remote_path_provider(label, remote_tp_ns);
+
+    // Upload manifest to cloud storage to mark it as mountable.
+    const auto upload_result = co_await _remote.upload_manifest(
+      _bucket, manifest, manifest.get_manifest_path(path_provider), parent);
+
+    if (upload_result != upload_result::success) {
+        vlog(
+          cst_log.debug,
```

Comment:
Let's log these as errors, or at least warnings

Reply:
Updated these `vlog` statements to error level logs.